### PR TITLE
Add reserved instances check

### PR DESCRIPTION
**Pull request by obazoud**
## Pull Request Checklist

**Is this in reference to an existing issue?**

#### General

- [x] Update Changelog following the conventions laid out on [Keep A Changelog](http://keepachangelog.com/)

- [x] Update README with any necessary configuration snippets

- [x] Binstubs are created if needed

- [x] RuboCop passes

- [x] Existing tests pass 

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

### CODE REVIEW

File: bin/check-reserved-instances.rb [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#   Copyright (c) 2015, Olivier Bazoud, olivier.bazoud@gmail.com
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk'
+
+class CheckReservedInstances < Sensu::Plugin::Check::CLI
+  option :aws_access_key,
```
Comment (sstarcher) [inline bin/check-reserved-instances.rb]: remove

File: bin/check-reserved-instances.rb [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+         short: '-a AWS_ACCESS_KEY',
+         long: '--aws-access-key AWS_ACCESS_KEY',
+         description: "AWS Access Key. Either set ENV['AWS_ACCESS_KEY'] or provide it as an option",
+         default: ENV['AWS_ACCESS_KEY']
+
+  option :aws_secret_access_key,
```
Comment (sstarcher) [inline bin/check-reserved-instances.rb]: remove

File: bin/check-reserved-instances.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+         description: "AWS Secret Access Key. Either set ENV['AWS_SECRET_KEY'] or provide it as an option",
+         default: ENV['AWS_SECRET_KEY']
+
+  option :aws_region,
+         short: '-r AWS_REGION',
+         long: '--aws-region REGION',
+         description: 'AWS Region (defaults to us-east-1).',
+         default: 'us-east-1'
+
+  option :use_iam_role,
```
Comment (sstarcher) [inline bin/check-reserved-instances.rb]: remove

File: bin/check-reserved-instances.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+         proc: proc(&:to_i)
+
+  option :critical,
+         description: 'Critical if expire date is lower age in seconds',
+         short: '-c SECONDS',
+         long: '--critical SECONDS',
+         default: 60 * 60 * 24 * 30 * 2,
+         proc: proc(&:to_i)
+
+  def aws_config
```
Comment (sstarcher) [inline bin/check-reserved-instances.rb]: remove

File: bin/check-reserved-instances.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    { access_key_id: config[:aws_access_key],
+      secret_access_key: config[:aws_secret_access_key],
+      region: config[:aws_region]
+    }
+  end
+
+  def run
+    aws_config = {}
+    if config[:use_iam_role].nil?
```
Comment (sstarcher) [inline bin/check-reserved-instances.rb]: remove

File: bin/check-reserved-instances.rb [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+# NOTES:
+#
+# LICENSE:
+#   Copyright (c) 2015, Olivier Bazoud, olivier.bazoud@gmail.com
+#   Released under the same terms as Sensu (the MIT license); see LICENSE
+#   for details.
+#
+
+require 'sensu-plugin/check/cli'
+require 'aws-sdk'
```
Comment (sstarcher) [inline bin/check-reserved-instances.rb]: include Common and follow the pattern of 

https://github.com/sensu-plugins/sensu-plugins-aws/blob/master/bin/check-ebs-snapshots.rb

**Comment by obazoud** [comment]
Thanks for the feedback @sstarcher.
I will update this PR.

**Comment by obazoud** [comment]
@sstarcher PR updated!

**Comment by sstarcher** [comment]
Thanks @obazoud

**Pull request merged**